src/main/scala/sigmastate/lang/syntax/Identifiers.scala

```diff
@@ -6,8 +6,10 @@ import sigmastate.lang.syntax.Basic._
 
 //noinspection ForwardReference
 object Identifiers {
-  case class NamedFunction(f: Char => Boolean)
-      (implicit name: sourcecode.Name) extends (Char => Boolean){
+  case class NamedFunction(f: Char => Boolean,
+                           name: sourcecode.Name = sourcecode.Name.generate)
+    extends (Char => Boolean){
+
```

Comment:
@ergomorphic I'm not sure the default value is much better here.

Reply:
Don’t think we need to change this at all. This issue relate to IR nodes only. Those that are subject to rewriting and thus need structural equality.